test/src/unit-alt-string.cpp

```diff
@@ -60,17 +66,36 @@ class alt_string
     }
 
     template <typename op_type>
-    bool operator==(op_type&& op) const
+    typename std::enable_if< // disable for alt_string
+                !std::is_same<  alt_string,
+                                typename std::remove_reference<op_type>::type
+                             >::value,
+                bool>::type
```

Comment:
I think you can remove all of this and simply take the parameter as `const op_type&`.

Reply:
Yeap, you are right, thanx for the notice. I will make another request.

Reply:
No need to open a different PR by the way, pushing another commit will serve the same purpose.

Reply:
Oh, I did not know, thanx.